### PR TITLE
GSP-93: Add ObjectMode Pair

### PR DESCRIPTION
**Pull request by Xuanwo**
Signed-off-by: Xuanwo <github@xuanwo.io>

### CODE REVIEW

**Review by JinnyYi — APPROVED** [review]

File: rfcs/93-add-object-mode-pair.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+- `Create`: set the output object's `ObjectMode`
+- `Stat`: ObjectMode hint, returns error if `ObjectMode` not meet.
+- `Delete`: ObjectMode hint
+
+For `Stat` and `Delete`
+
+- Service SHOULD use the `ObjectMode` pair as a hint.
+- Service could have different implementations for different `ObjectMode`.
+
+Take `s3` as an example, we simulate `CreateDir` via creating object ends with `/`. `CreateDir("test")` will create an object `test/` in s3. And we can
```
Comment (Prnyself) [inline rfcs/93-add-object-mode-pair.md:47]: So, is `CreateDir("test")` the same as `CreateDir("test/")`? Or shall we declare that no `/` should be added we `CreateDir` was called?

Comment (JinnyYi) [reply rfcs/93-add-object-mode-pair.md:47]: For consistency, we don't allow user input path ends with `/` for all API.

Comment (Xuanwo) [reply rfcs/93-add-object-mode-pair.md:47]: Yes, `CreateDir("test/")` is not allowed.

**Comment by Xuanwo** [comment]
ping @Prnyself to give me another LGTM.

**Review by Prnyself — APPROVED** [review]
LGTM

**Pull request merged**